Ecommerce project

ZAINABFATIMA0 opened the pull request:
E-commerce app for products listing

ZAINABFATIMA0 committed:
Added .gitignore files

ZAINABFATIMA0 committed:
Ecommerce Project
In ecommerce project ecommerce_app is listing all the products right now
Two views are created for product's list and product's details

sheralim012 commented on Ecommerce/.gitignore:
read about django project and app naming convention
we normally don't use capital characters in project and app name

sheralim012 replied:
ecommerce app is too broad

sheralim012 commented on Ecommerce/Ecommerce/urls.py:
```diff
@@ -0,0 +1,7 @@
+from django.urls import include, path
+
+urlpatterns = [
+    path("ecommerce_app/", include("ecommerce_app.urls")),
```
```suggestion
    path("ecommerce/", include("ecommerce.urls")),
```

sheralim012 commented on Ecommerce/ecommerce_app/models.py:
```diff
@@ -0,0 +1,25 @@
+
+
+class Product(models.Model):
+    product_name = models.CharField(max_length = 255)
```
```suggestion
    product_name = models.CharField(max_length=255)
```

read about pep8 and django formatting, there is a package named black that handles these formatting changes

sheralim012 replied:
```suggestion
    name = models.CharField(max_length=255)
```

no need to add product prefix as model name is product as well

sheralim012 commented on Ecommerce/ecommerce_app/models.py:
```diff
@@ -0,0 +1,25 @@
+    sku = models.CharField(max_length = 50, unique = True)
+    price = models.JSONField()
+    description = models.TextField()
+    details = models.JSONField(blank = True, null = True)
```
would it be better to have a separate model to store details as this violates 1 normal form

sheralim012 commented on Ecommerce/ecommerce_app/management/commands/parse_productdata.py:
```diff
@@ -0,0 +1,36 @@
+    def add_arguments(self, parser):
+        parser.add_argument('json_file', type=str, help='The JSON file containing products')
+
+    def load_data_from_json(self, json_file):
```
I would suggest to use factory-boy to create data

ZAINABFATIMA0 committed:
Made changes in model, naming, PEP-08 compliance
Added models to ensure they adhere to 1NF
Changed name of apps to more descriptive names
Made sure code adheres to PEP-08 rules
Created dummy data to display using factory-boy

ZAINABFATIMA0 committed:
Removed Ecommerce

sheralim012 commented on ecommerce/clothing_catalogue/urls.py:
```diff
@@ -0,0 +1,8 @@
+from . import views
+
+urlpatterns = [
+    path('products/', views.product_list, name='product-list'),
```
```suggestion
    path('products/', views.product_list, name='product_list'),
```

urls have dashes `-` and URL names have underscores

sheralim012 commented on ecommerce/clothing_catalogue/models.py:
```diff
@@ -0,0 +1,41 @@
+    duppata = models.TextField(default=None)
+
+
+class Additional_attributes(models.Model):
```
classes have PascalCase name and usually singular 
```suggestion
class Attribute(models.Model):
```

sheralim012 commented on ecommerce/clothing_catalogue/models.py:
```diff
@@ -0,0 +1,41 @@
+    
+    shirt = models.TextField(default=None)
+    trouser = models.TextField(default=None)
+    duppata = models.TextField(default=None)
```
not every peice of clothing will have every thing also instead of setting default as none user nullable and/or blank fields

sheralim012 commented on ecommerce/clothing_catalogue/management/commands/parse_productdata.py:
```diff
@@ -0,0 +1,39 @@
+        json_file = kwargs['json_file']
+        with open(json_file, 'r') as file:
+            products = json.load(file)
+            for product_data in products:
```
discuss with Hamza about using factory boy classes to create objects

ZAINABFATIMA0 committed:
Removed parse_productdata

ZAINABFATIMA0 committed:
Used factory-boy to create dummy data

ZAINABFATIMA0 committed:
Final Commit
